hibernate-core/src/main/java/org/hibernate/dialect/OracleDialect.java

```diff
@@ -643,6 +646,17 @@ public void contributeTypes(TypeContributions typeContributions, ServiceRegistry
 
 			typeContributions.contributeSqlTypeDescriptor( descriptor );
 		}
+
+		// We need a special type descriptor for Oracle, because that must return BIT as SQL type
+		// in order for PreparedStatement#setNull to work properly
+		typeContributions.contributeSqlTypeDescriptor(
```

Comment:
I really think we should try `Types#NULL` for the cases where we need to set null values.  No idea if that works for all drivers/databases but if so its a much better option IMO than randomly picking one

Reply:
The other option is to delegate binding nulls to the `WrapperOptions` object. It's what it exists for.

We could either let `WrapperOptions` remap "null" JDBC type codes, or we could let it actually handle the whole call to `setNull()`.

Reply:
Or even delegate back to Dialect through WrapperOptions

Reply:
Yep, right, that's what I mean. Should have been more explicit.